src/axom/primal/geometry/Polyhedron.hpp

```diff
@@ -73,7 +73,9 @@ class NeighborCollection
    */
   AXOM_HOST_DEVICE int getNumNeighbors(int vtx) const
   {
+#if !defined(__HIP_DEVICE_COMPILE__)
     SLIC_ASSERT(vtx >= 0 && vtx < MAX_VERTS);
+#endif
```

Comment:
Minor:  Not strictly necessary for this PR, but is there a way to fold the `#if defined()` into `SLIC_ASSERT`? That way developers won't need to be aware of this restriction.